packages/express/src/middlewares/authenticateUser.ts

```diff
@@ -110,62 +111,76 @@ const middlewareGetterByAuthType: Record<
-
       try {
-        const authAttempts = await Promise.all(
-          authMethods.map((method) =>
-            middlewareGetterByAuthType[method](
-              method === AuthMethods.JWT ? jwt : req.headers.authorization,
-              options,
-              logger
-            )
-          )
-        );
-
-        const successfulAuthAttempts = compact(authAttempts);
-
-        if (isEmpty(successfulAuthAttempts)) {
+        await authProvidersMiddleware(req, options, authMethods);
+        next();
+      } catch (error: unknown) {
+        if (error instanceof AuthenticationMethodError) {
           res
-            .status(401)
-            .json(
-              render401Error([
-                authAttempts.includes(null)
-                  ? `Token invalid for methods ${authMethods}`
-                  : `Token not set for methods ${authMethods}`,
-              ])
-            );
+            .status(error.httpCode())
+            .json(render401Error(error.requiredChecks));
           return;
         }
-
-        const jwtUser: JWTUser = successfulAuthAttempts[0];
-        req.context.jwtUser = jwtUser;
-        req.context.authenticated.push(
-          jwtUser.employeeReference !== "0" ? "jwt" : "api_key"
-        );
-        next();
-      } catch (error: unknown) {
-        logger.info("auth.authenticateUser.error", {
-          jwt,
-          token: req.headers.authorization,
-          methods: AuthMethods,
-          error,
-        });
         res.status(401).json(render401Error(authMethods));
       }
     };
   };
 }
+
+export async function authProvidersMiddleware<Options extends AuthOptions>(
+  req: GenericRequest,
+  options: Options,
+  authMethods: AuthMethods[],
+): Promise<JWTUser> {
+  const logger = getRequestLogger(req);
+  const cookies = (req.cookies as Record<string, string | undefined>) || {};
+  const jwt =
+    cookies.jwt || parseAuthorizationHeader(req.headers.authorization);
+
+  try {
+    const authAttempts = await Promise.all(
+      authMethods.map((method) =>
+        middlewareGetterByAuthType[method](
+          method === AuthMethods.JWT ? jwt : req.headers.authorization,
+          options,
+          logger,
+        ),
+      ),
+    );
+
+    const successfulAuthAttempts = compact(authAttempts);
+
+    if (isEmpty(successfulAuthAttempts)) {
+      throw new AuthenticationMethodError([
+        authAttempts.includes(null)
+          ? `Token invalid for methods ${authMethods}`
+          : `Token not set for methods ${authMethods}`,
+      ]);
+    }
+
+    const jwtUser: JWTUser = successfulAuthAttempts[0];
+    req.context.jwtUser = jwtUser;
+    req.context.authenticated.push(
+      jwtUser.employeeReference !== "0" ? "jwt" : "api_key",
+    );
+    return jwtUser;
+  } catch (error: unknown) {
+    logger.info("auth.authenticateUser.error", {
+      jwt,
+      token: req.headers.authorization,
+      methods: AuthMethods,
+      error,
+    });
+    throw error;
```

Comment:
juli, que te parece algo asi? mas que nada porq si falla cualquier otra cosa, va a estar tirando un error general que va a devolver 500 y eso no lo estamos catcheando
Creo que seria
```suggestion
if (error instanceof AuthenticationMethodError) {
  throw error;
}
else {
   throw new AuthenticationMethodError(authMethods)
}
```